python/triton/language/semantic.py

```diff
@@ -1306,7 +1309,7 @@ def dot(lhs: tl.tensor,
     def assert_dtypes_valid(lhs_dtype, rhs_dtype, arch):
         if is_hip():
             assert lhs.dtype == rhs.dtype or (lhs.type.scalar.is_fp8() and rhs.type.scalar.is_fp16()) or \
-                (lhs.type.scalar.is_fp16() and rhs.type.scalar.is_fp8()), \
+                (lhs.type.scalar.is_fp16() and rhs.type.scalar.is_fp8()) or (lhs.type.scalar.is_fp8() and rhs.type.scalar.is_fp8()), \
```

Comment:
is this (lhs.is_fp8 and rhs.is_fp8) included in lhs.dtype == rhs.dtype?
And MI300 mfma also supports fp8/bf8 and bf8/fp8. Do we need to add those here?

Reply:
is_fp8() is true for both fp8 and bf8(), but they are not equal.

Reply:
That makes sense.